packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp

```diff
@@ -1293,7 +1300,11 @@ class Impl {
       if ( ! force && (sweep - 1) % sweep_step_) return false;
       if (collective_) {
 #ifdef HAVE_MPI
```

Comment:
See above.  `Tpetra::idot` and `Tpetra::Details::iallreduce` both return an `std::shared_ptr` to an object with a `wait()` method.

Reply:
@mhoemmen, sounds good.

Reply:
FYI the fall-back for `MPI_VERSION < 3` works by deferring a blocking all-reduce until you call `wait()`.  This means that calling `Tpetra::Details::iallreduce` or `Tpetra::idot` is always nonblocking.